src/app/security/maturity-model/components/Article.tsx

```diff
@@ -0,0 +1,50 @@
+'use client'
+
+import Link from 'next/link'
+
+import { LinkSimple } from '@phosphor-icons/react/dist/ssr'
+import type { Route } from 'next'
+import { useIntersectionObserver } from 'usehooks-ts'
+
+import { Icon } from '@/components/Icon'
+
+import {
+  useSectionHashObserver,
+  HASH_SIGN,
+} from '../utils/useSectionHashObserver'
+
+type LinkTitleProps = {
+  title: string
+  slug: string
+} & React.ComponentPropsWithoutRef<'article'>
+
+export function Article({ title, slug, children }: LinkTitleProps) {
+  const { updateSectionHash } = useSectionHashObserver()
+
+  const { ref } = useIntersectionObserver({
+    onChange: (isIntersecting) => {
+      if (isIntersecting) {
+        updateSectionHash(slug)
+      }
+    },
+  })
+
+  const sectionHash = HASH_SIGN + slug
+
+  return (
+    <article ref={ref}>
+      <Link
+        href={sectionHash as Route}
+        className="group text-brand-100 hover:no-underline"
+      >
+        <h3 id={slug} className="flex items-center gap-2">
+          {title}
+          <span className="invisible group-hover:visible">
+            <Icon component={LinkSimple} size={18} />
+          </span>
+        </h3>
+      </Link>
+      {children}
+    </article>
```

Comment:
We should avoid wrapping heading elements in links, as this compromises their semantic purpose. Headings are meant to structure content and aid navigation for both screen readers and search engines.

n addition, wrapping the entire heading in a link can lead to keyboard accessibility issues, where these links might be skipped over or incorrectly focused – as is currently the case.

What about something like this?

```suggestion
    <article ref={ref}>
      <h3 id={slug}>
        <Link
          href={sectionHash as Route}
          className="group inline-flex items-center gap-2 text-brand-100 hover:no-underline"
        >
          {title}
          <span className="invisible group-hover:visible">
            <Icon component={LinkSimple} size={18} />
          </span>
        </Link>
      </h3>
      {children}
    </article>
```

Reply:
I wasn't aware of that, thanks for sharing :)